src/sherpa_ai/tools.py

```diff
@@ -202,3 +211,43 @@ def _run(self, query: str) -> str:
 
     def _arun(self, query: str) -> str:
         raise NotImplementedError("UserInputTool does not support async run")
+
+class HugChatTool(BaseTool):
+    name = "Hugchat"
+    description = (
+        "Access the user input for the task."
+        "You use this tool if you need to use HugChat for Q&A."
+    )
+    def _run(self, query: str) -> str:
+
+        email = cfg.HUGCHAT_EMAIL
+        passwd = cfg.HUGCHAT_PASS
+        # Log in to huggingface and grant authorization to huggingchat
+        sign = Login(email, passwd)
+        cookies = sign.login()
+
+        # Save cookies to the local directory
+        cookie_path_dir = "./cookies_snapshot"
+        sign.saveCookiesToDir(cookie_path_dir)
+
+        # Create a ChatBot
+        chatbot = hugchat.ChatBot(cookies=cookies.get_dict())  # or cookie_path="usercookies/<email>.json"
+
+        query_result = ""
+
+        if cfg.HUGCHAT_CONFIG == 1:
```

Comment:
I suggest using a `case/match` statement here instead of multiple `if` options, since they are mutually exclusive.